src/ressa/context.rs

```diff
@@ -10,31 +10,34 @@ const RESOLVES_TO: &'static str = "???";
 const TRANSIENT: &'static str = "";
 
 /// Context used by the Parser, storing local variables (#{varname}) and objects/tags
-#[derive(Default, Debug, Clone, Any, PartialEq)]
+#[derive(Default, Debug, Clone, Any)]
 pub struct ParserContext {
-    variables: ContextData,
-    local_variables: HashMap<String, String>,
+    objectlike_data: HashMap<String, Object>,
+    variables: HashMap<String, String>,
     pub frame_number: i32,
 }
 
-impl Into<ContextData> for ParserContext {
-    fn into(self) -> ContextData {
-        self.variables
+impl Into<RessaResult> for ParserContext {
+    fn into(self) -> RessaResult {
+        self.objectlike_data
             .into_iter()
             .filter(|(_, val)| !val.contains_key(RESOLVES_TO) && !val.contains_key(TRANSIENT))
+            .map(|obj| (obj.0, obj.1.into_inner()))
             .collect()
     }
 }
 
 /// Interface of the Context, offering ability to create, read, and update objects/tags
 pub trait ContextObjectActions {
     fn make_object(&mut self, name: &str);
+    fn save_object(&mut self, name: &str, new_obj: &Object);
+    /// Legacy API for generating object attributes. TBD if we keep
     fn make_attribute(&mut self, name: &str, attr_name: &str, attr_value: Option<String>);
```

Comment:
Yeah I think we probably don't need this anymore. We should be able to modify the rune Objects to add attributes to one